pyvista/core/filters/data_set.py

```diff
@@ -4906,7 +4906,7 @@ def merge(
         _update_alg(append_filter, progress_bar, 'Merging')
         merged = _get_output(append_filter)
         if inplace:
-            if type(self) == type(merged):
+            if isinstance(self, type(merged)):
                 self.deep_copy(merged)
```

Comment:
The original logic seems to be that the types have to be the same.  The error message in this case says to use `is` comparison in this case.  I haven't checked whether this change makes more sense though.

Reply:
Thanks for your comment, I am looking into it as I don't understand what the difference is between the two methods.

Reply:
Ah, I see. [isinstance](https://docs.python.org/3/library/functions.html#isinstance) includes the inherited class as well, so a comparison by is is appropriate, as you say.